src/MSBuild.Conversion.SDK/SemanticVersionComponentizedString.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+
+namespace MSBuild.Conversion
+{
+    public class SemanticVersionComponentizedString
+    {
+        public SemanticVersionComponentizedString(string rawString)
+        {
+            _rawString = rawString ?? string.Empty;
+        }
+
+        private readonly string _rawString;
+        private IReadOnlyList<MixedTypeComparisonToken> _parts;
+
+        public IReadOnlyList<MixedTypeComparisonToken> Parts
+        {
+            get
+            {
+                EnsureParsed();
+                return _parts;
+            }
+        }
+
+        private static Regex _partSplitterRegex = new Regex(@"
+            (\.)            # literal dot - parenthesised so the dot tokens are included in the split results.
+            |               # The dot MUST be above the zero-width assertions, otherwise it will not necessarily be split / captured.
+            (?<=\d)(?=\D)   # digit / non-digit boundary
+            |
+            (?<=\D)(?=\d)   # non-digit / digit boundary
+            ", RegexOptions.IgnorePatternWhitespace);
+
+        private void EnsureParsed()
+        {
+            if (_parts == null)
```

Comment:
```suggestion
            if (_parts is null)
```

we use `is null` in this house